src/pages/Facility/settings/locations/LocationMap.tsx

```
@@ -0,0 +1,610 @@
+        let current = location;
+        while (current.parent?.id) {
+          matchedIds.add(current.parent.id);
+          const parent = locations.find((loc) => loc.id === current.parent?.id);
+          if (!parent) break;
+          current = parent;
+        }
+      }
+    });
+
+    // Expand all nodes when there's a search query, collapse all when search is cleared
+    if (searchQuery && searchQuery.trim()) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)